beta-src/src/state/game/game-api-slice.ts

```diff
@@ -293,13 +305,116 @@ const drawBuilds = (state) => {
   }
 };
 
+const updateUnitsDisbanding = (state) => {
+  const {
+    data: {
+      data: { contextVars, currentOrders, units },
+    },
+    territoriesMeta,
+    ordersMeta,
+  }: {
+    data;
+    territoriesMeta;
+    ordersMeta;
+  } = current(state);
+  if (currentOrders && contextVars) {
+    const { context } = contextVars;
+    if (context.phase === "Retreats") {
+      const tMeta: TerritoriesMeta = territoriesMeta;
+      const terrMetaEntries = Object.entries(tMeta);
+      const overtakenTerritories = terrMetaEntries.filter(([id, val]) => {
+        return (
+          val.countryID === context.countryID &&
+          val.countryID !== val.ownerCountryID
+        );
+      });
+
+      let userDisbandingUnits;
+      let territoryKeys;
+      console.log("ot", overtakenTerritories);
+
+      if (overtakenTerritories && overtakenTerritories.length > 0) {
+        const terrKeyTempArray: string[] = [];
+        const disbandingUnitsTempArray: CurrentOrder[] = [];
+        overtakenTerritories.forEach(([terrKey, val]) => {
+          const { unitID } = val;
+          console.log("OM", ordersMeta);
+          const overtakenUnitsInOrder = currentOrders.find(
+            (o) =>
+              (val.id === units[o.unitID].terrID &&
+                o.unitID !== unitID &&
+                o.type === "Disband") ||
+              (val.id === units[o.unitID].terrID &&
+                o.unitID !== unitID &&
+                ordersMeta[o.id] &&
+                ordersMeta[o.id].update.type === "Disband"),
+          );
+
+          console.log("otuio", overtakenUnitsInOrder);
+
+          if (overtakenUnitsInOrder) {
+            disbandingUnitsTempArray.push(overtakenUnitsInOrder);
+            terrKeyTempArray.push(terrKey);
+          }
+        });
+        territoryKeys = terrKeyTempArray;
+        userDisbandingUnits = disbandingUnitsTempArray;
+      }
+      if (!userDisbandingUnits) {
+        userDisbandingUnits = currentOrders;
+      }
+      console.log("userDisbandingUnits", userDisbandingUnits);
+      if (userDisbandingUnits) {
```

Comment:
this will always be true right? You're checking to see if it is undefined on line 363 and making it equivalent to `currentOrders` if it isn't.

Reply:
Would current orders ever be empty at any point?

Reply:
yup, but in that case you shouldn't assign it